### ISSUE TITLE
ImGui assertion fails at resize event on macOS 

### ISSUE THREAD

**Issue by miko007** [cinder/Cinder#2173]
If one resizes the window on macOS and *ImGui* is used, the cinder app crashes, because the assertion `g.WithinFrameScope` fails. To fix this, line `528` *(commit cd1d70191a4a62e9a303c620271e0341758cdf04)* in `src/cinder/CinderImGui.cpp` 

https://github.com/cinder/Cinder/blob/cd1d70191a4a62e9a303c620271e0341758cdf04/src/cinder/CinderImGui.cpp#L528 hast to be changed to

```c++
    sWindowConnections[window] += window->getSignalDraw().connect( std::bind( ImGui_ImplCinder_NewFrameGuard, window ) );
```

As of the huge backlog of pull requests, i will not open another one. Maybe one of the developers can implement this little fix directly.

**Comment by num3ric** [comment]
Hi @miko007, thanks for the report. Unfortunately, I don't have access to a mac with Xcode to replicate this issue at the moment.

However, using the window's draw signal would prevent the use of `Dear ImGui` in the app `update()` function (a common pattern for small apps/samples). Using `ImGui::` functions there would then also result in the same `g.WithinFrameScope` assertion error.

A platform-specific fix may instead affect the `ImGui_ImplCinder_Resize` or the `ImGui_ImplCinder_NewFrameGuard` functions, but hard to tell from afar.

**Comment by miko007** [comment]
that is correct, while i am not shure why one could not do the gui in `App::draw()` instead, because thats what it is for, right?

But if that is a concern, i am not really shure how to proceed from here, as *ImGui* is pretty much not usable on macOS without the fix...

**Comment by davidgranstrom** [comment]
I also noticed this issue on macOS when trying out Cinder. I rebuilt library with the patch from #2178 and resizing the window (with ImGui) seem to work fine now!
